### PR TITLE
feature: add PodConversion API

### PR DESCRIPTION
**Pull request by duanmengkk**
#### What type of PR is this?

/kind feature

#### Which issue(s) does this PR fix?

Fixes parts of #307
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/apis/kosmos/v1alpha1/podconvertpolicy_types.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	// +optional
+	Spec PodConvertPolicySpec `json:"spec,omitempty"`
+}
+
+type PodConvertPolicySpec struct {
+	// A label query over a set of resources.
+	// If name is not empty, labelSelector will be ignored.
+	// +required
+	LabelSelector *metav1.LabelSelector `json:"labelSelector"`
+
```
Comment (OrangeBao) [inline pkg/apis/kosmos/v1alpha1/podconvertpolicy_types.go:27]: We should refer to this: [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required)

File: pkg/apis/kosmos/v1alpha1/podconvertpolicy_types.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PodConvertPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification for the behaviour of the podConversion.
+	// +optional
+	Spec PodConvertPolicySpec `json:"spec,omitempty"`
+}
```
Comment (OrangeBao) [inline pkg/apis/kosmos/v1alpha1/podconvertpolicy_types.go:20]: ```Spec``` is required, remove the ```+optional``` ?

Comment (duanmengkk) [reply pkg/apis/kosmos/v1alpha1/podconvertpolicy_types.go:20]: OK

**Pull request merged**